ludwig/datasets/mixins/download.py

```diff
@@ -165,3 +212,33 @@ def download_raw_dataset(self):
     @property
     def download_url(self):
         return self.config["download_urls"]
+
+
+class S3FileDownloadMixin:
```

Comment:
No need to use S3 specific stuff, we can just use `fsspec`, which abstracts away these S3-like filesystems. See `fs_utils` in Ludwig for examples.

Reply:
Done!